### PR TITLE
Chapter 8 : Basic Login

### PR DESCRIPTION
**Pull request by DaudKhadijah01**
ready

### CODE REVIEW

**Comment by DaudKhadijah01** [comment]
ready

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -2,12 +2,33 @@
...
-    <%= link_to "sample app", "#", id: "logo" %>
+    <%= link_to t(:title), home_path, id: "logo" %>
     <nav>
       <ul class="nav navbar-nav navbar-right">
-         <li><%= link_to "Home", "#" %></li>
-         <li><%= link_to "Help", "#" %></li>
-         <li><%= link_to "Log in", "#" %></li>
+         <li><%= link_to t(:home), home_path %></li>
+         <li><%= link_to t(:help), help_path %></li>
+         <li><%= link_to t(:contact), contact_path%></li>
```
Comment (ghost) [inline app/views/layouts/_header.html.erb]:
```suggestion
         <li><%= link_to t(:contact), contact_path %></li>
```

File: app/assets/stylesheets/sessions.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
```
Comment (ghost) [inline app/assets/stylesheets/sessions.scss]: delete this file, please

File: test/controllers/sessions_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+require "test_helper"
```
Comment (ghost) [inline test/controllers/sessions_controller_test.rb]: delete this file, please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by DaudKhadijah01** [comment]
ready

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -2,11 +2,13 @@ class ApplicationController < ActionController::Base
...
   protect_from_forgery with: :exception
   before_action :set_locale
+  include SessionsHelper
 
   private
 
   def set_locale
-   locale = params[:locale] || I18n.default_locale
+   locale = params[:locale].to_s.strip.to_sym
+    I18n.locale = I18n.available_locales.include?(locale) ? locale : I18n.default_locale
```
Comment (ngocvt-0484) [inline app/controllers/application_controller.rb:11]: wrong indent

File: config/routes.rb [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
   scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
-    get 'home/index'
-    resources :posts
-    root 'static_pages#home'
-    get 'static_pages/help'
-    get 'static_pages/contact'
+
+    root :to => 'static_pages#home', as: 'home'
+    get '/help' => 'static_pages#help', as: 'help'
+    get '/contact' => 'static_pages#contact', as: 'contact'
```
Comment (ngocvt-0484) [inline config/routes.rb]:
```suggestion
    get "/contact", to: "static_pages#contact", as: "contact"
```

**Comment by DaudKhadijah01** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 6]
```diff
@@ -13,7 +13,7 @@ textarea {
 }
 
 .center {
-  text-align: center;
+  text-allign: center;
```
Comment (huongnt-2545) [inline app/assets/stylesheets/custom.scss]:
```suggestion
  text-align: center;
```

**Comment by DaudKhadijah01** [comment]
ready

File: config/routes.rb [hunk 7]
```diff
@@ -1,12 +1,16 @@
...
-    get 'static_pages/contact'
+
+    root :to => 'static_pages#home', as: 'home'
+    get "/help", to: "static_pages#help", as: "help"
+    get "/contact", to: "static_pages#contact", as: "contact"
+
     get "/signup", to: "users#new"
     post "/signup", to: "users#create"
-    resources :users, only: %i(new create show)
+    resources :users, only: %i(show new create destroy)
```
Comment (ghost) [inline config/routes.rb]: why you have a route for destroy action?

**Review by ghost — COMMENTED** [review]
cmt

File: config/routes.rb (hunk 7 above)
Comment (DaudKhadijah01) [reply config/routes.rb]: i just follow coding in the video. my other friends also include destroy. should i delete it?

**Comment by DaudKhadijah01** [comment]
ready

**Comment by DaudKhadijah01** [comment]
why account does not appear on my header?
image

File: config/routes.rb [hunk 8]
```diff
@@ -1,12 +1,16 @@
...
   scope "(:locale)", locale: /#{I18n.available_locales.join("|")}/ do
-    get 'home/index'
-    resources :posts
-    root 'static_pages#home'
-    get 'static_pages/help'
-    get 'static_pages/contact'
+
+    root :to => 'static_pages#home', as: 'home'
+    get "/help", to: "static_pages#help", as: "help"
+    get "/contact", rato: "static_pages#contact", as: "contact"
```
Comment (ngocvt-0484) [inline config/routes.rb]:
```suggestion
    get "/contact", to: "static_pages#contact", as: "contact"
```

**Comment by DaudKhadijah01** [comment]
ready

File: app/views/sessions/create.html.erb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+<h1>Sessions#create</h1>
```
Comment (thaodtd-0665) [inline app/views/sessions/create.html.erb:1]: remove this file

File: app/views/sessions/destroy.html.erb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+<h1>Sessions#destroy</h1>
```
Comment (thaodtd-0665) [inline app/views/sessions/destroy.html.erb:1]: remove this file

**Comment by DaudKhadijah01** [comment]
can i merge after this? ready

**Comment by DaudKhadijah01** [comment]
ready

**Review by thaodtd-0665 — APPROVED** [review]

**Comment by thaodtd-0665** [comment]
merged

**Pull request merged**